### PR TITLE
Add exponential backoff functionality to circuit breakers

### PR DESCRIPTION
**Pull request by mlegore**
I added functionality to introduce variable wait times for Circuit breakers in the open state. The wait time can be configured using an IntervalFunction (as is used in resilience-retry). This is functionality we needed within our organization, that I thought might be useful to others.

### CODE REVIEW

**Comment by RobWin** [comment]
Thx, could you please create an issue as well and describe your solution?

**Comment by RobWin** [comment]
Do you need an option to reset the attempts counter? 
For example after 10 attempts, reset to the initial wait interval? Or should this be the task of the intervalFunction?

**Comment by RobWin** [comment]
Is the attempts counter set to 0 again, when the CircuitBreaker was closed after it was half open?

**Comment by RobWin** [comment]
Yes, the counter is set to 0. Very good.

File: resilience4j-circuitbreaker/src/main/java/io/github/resilience4j/circuitbreaker/CircuitBreakerConfig.java [hunk 1]
```diff
@@ -60,6 +61,7 @@
     private boolean writableStackTraceEnabled = DEFAULT_WRITABLE_STACK_TRACE_ENABLED;
```
Comment (RobWin) [inline resilience4j-circuitbreaker/src/main/java/io/github/resilience4j/circuitbreaker/CircuitBreakerConfig.java]: We could remove `waitDurationInOpenState`  and replace it by a constant waitIntervalFunction

File: resilience4j-circuitbreaker/src/main/java/io/github/resilience4j/circuitbreaker/CircuitBreakerConfig.java [hunk 2]
```diff
@@ -60,6 +61,7 @@
...
     private Duration waitDurationInOpenState = Duration.ofSeconds(DEFAULT_WAIT_DURATION_IN_OPEN_STATE);
```
Comment (RobWin) [inline resilience4j-circuitbreaker/src/main/java/io/github/resilience4j/circuitbreaker/CircuitBreakerConfig.java]: We could remove `waitDurationInOpenState` and replace it by using a constant default `waitIntervalFunctionInOpenState`

**Comment by mlegore** [comment]
https://github.com/resilience4j/resilience4j/issues/671 created issue

File: resilience4j-circuitbreaker/src/main/java/io/github/resilience4j/circuitbreaker/CircuitBreakerConfig.java (hunk 2 above)
Comment (mlegore) [reply resilience4j-circuitbreaker/src/main/java/io/github/resilience4j/circuitbreaker/CircuitBreakerConfig.java]: That would be a breaking change to remove `getWaitDurationInOpenState()` from this class

Comment (mlegore) [reply resilience4j-circuitbreaker/src/main/java/io/github/resilience4j/circuitbreaker/CircuitBreakerConfig.java]: I can do that if you want, up to you

Comment (RobWin) [reply resilience4j-circuitbreaker/src/main/java/io/github/resilience4j/circuitbreaker/CircuitBreakerConfig.java]: No, its fine. I can adjust it.

**Comment by mlegore** [comment]
@RobWin do you have more changes you need? or did I miss something

**Comment by storozhukBM** [comment]
I also did a quick review, and it looks good for me.
Unfortunately, I don't see the results of static code analysis and test coverage.
I am going to kick TravisCI once again.

**Comment by RobWin** [comment]
@mlegore No, I will merge it when I'm back from vacation. I will add my changes after it has been merged.

**Comment by RobWin** [comment]
@mlegore it would be nice if we could add this feature to the Spring Boot modules as well.

**Pull request merged**